proposals.go

```diff
@@ -0,0 +1,93 @@
+package dcrlibwallet
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	proposalsEndpoint = "https://proposals.decred.org/api/v1/proposals/vetted"
+	policyEndpoint    = "https://proposals.decred.org/api/v1/policy"
+)
+
+func (mw *MultiWallet) createHTTPClient() {
+	mw.httpClient = &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: 2,
+		},
+		Timeout: time.Duration(10) * time.Second,
+	}
+}
+
+func (mw *MultiWallet) get(url string, responseData interface{}) error {
+	res, err := mw.httpClient.Get(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	return json.NewDecoder(res.Body).Decode(responseData)
+}
+
+func (mw *MultiWallet) getPolicy() (*Policy, error) {
+	var policy *Policy
+	err := mw.get(policyEndpoint, &policy)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching politea policy: %v", err)
+	}
+
+	return policy, nil
+}
+
+// GetProposalsChunk gets proposals starting after the proposal with the specified
+// censorship hash. The number of proposals returned is specified in the poltiea
+// policy API endpoint
+func (mw *MultiWallet) GetProposalsChunk(from string) ([]Proposal, error) {
+	var result Proposals
+	url := proposalsEndpoint
+	if from != "" {
+		url = fmt.Sprintf(proposalsEndpoint+"?after=%s", from)
+	}
+
+	err := mw.get(url, &result)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching proposals from %s: %v", from, err)
+	}
+
+	return result.Proposals, nil
+
+}
+
+// GetAllProposals fetches all the proposals from the API
+func (mw *MultiWallet) GetAllProposals() ([]Proposal, error) {
```

Comment:
The return type of this function should be a json string of `[]Proposal` to make it visible after compiling with gomobile.